### PR TITLE
Add index entries for Modules 9 through 16.

### PR DESCRIPTION
**Pull request by luema**
Should \index[definitions]{...} commands be placed in the definitions.tex file? I placed them next to the \SavedDefinitionRender commands in the module#.tex files.

Also, I'm not sure why this is happening in module 9 (line 601):

### CODE REVIEW

File: book/modules/module12.tex [hunk 1]
```diff
@@ -21,7 +21,7 @@
 is invertible if there exists an inverse function that when composed with the original function
 produces the identity function and vice versa.
 
-\SavedDefinitionRender{InverseFunction}
+\SavedDefinitionRender{InverseFunction}\index[definitions]{Linear transformation!inverse}
```
Comment (siefkenj) [inline book/modules/module12.tex:24]: These are things about functions, don't restrict them to linear transformations.

File: book/modules/module12.tex [hunk 2]
```diff
@@ -73,7 +73,7 @@
 \]
 and so $\mathcal T$ has the same domain and codomain (at least a domain and codomain of the same dimension).
 
-Using the rank-nullity theorem, we can start developing a list of properties that are equivalent to invertibility of a
+Using the rank-nullity theorem\index{Linear transformation!rank-nullity theorem}, we can start developing a list of properties that are equivalent to invertibility of a
```
Comment (siefkenj) [inline book/modules/module12.tex:76]: Just reference `Rank-nullity theorem`

File: book/modules/module9.tex [hunk 3]
```diff
@@ -316,7 +316,7 @@
 The operation of applying a transformation to a specific set of vectors and seeing what results is
 called taking the \emph{image} of a set.
 
-\SavedDefinitionRender{ImageofaSet}
+\SavedDefinitionRender{ImageofaSet}\index[definitions]{Set!image}
```
Comment (siefkenj) [inline book/modules/module9.tex:319]: -> `Set!image of`

File: book/modules/module9.tex [hunk 4]
```diff
@@ -609,7 +609,7 @@
 \end{proof}
 
 \begin{theorem}
-	If $T:\R^n\to\R^m$ is a linear transformation, then $T$ takes lines to lines (or points).
+	If $T:\R^n\to\R^m$ is a linear transformation, then $T$ takes lines\index{Line!under a linear transformation} to lines (or points).
```
Comment (siefkenj) [inline book/modules/module9.tex:612]: remove

File: book/modules/module9.tex [hunk 5]
```diff
@@ -654,7 +654,7 @@
 \end{proof}
 
 \begin{theorem}
-	If $T:\R^n\to\R^m$ is a linear transformation, then $T$ takes subspaces to subspaces.
+	If $T:\R^n\to\R^m$ is a linear transformation, then $T$ takes subspaces\index{Subspace!under a linear transformation} to subspaces.
```
Comment (siefkenj) [inline book/modules/module9.tex:657]: Remove

File: book/modules/module11.tex [hunk 6]
```diff
@@ -513,7 +513,7 @@
 
 \medskip
 There is an equivalent Rank-nullity Theorem for linear transformations.
-\begin{theorem}(Rank-nullity Theorem for Linear Transformations) Let $\mathcal T$ be a linear transformation.
+\begin{theorem}(Rank-nullity Theorem for Linear Transformations)\index{Linear transformation!rank-nullity theorem} Let $\mathcal T$ be a linear transformation.
```
[the remark this answers is not in the document]
Comment (luema) [reply book/modules/module11.tex:516]: `\index{Rank-nullity theorem!for linear transformations}`

File: book/modules/module12.tex (hunk 1 above)
Comment (luema) [reply book/modules/module12.tex:24]: normal entry no sub

**Pull request closed without merging**